src/SimulationStrategyFactory.py

```diff
@@ -24,7 +26,8 @@ def __init__(self, experiment_configuration: BosonSamplingExperimentConfiguratio
             StrategyTypes.FIXED_LOSS: self.__generate_fixed_losses_strategy,
             StrategyTypes.UNIFORM_LOSS: self.__generate_uniform_losses_strategy,
             StrategyTypes.CLIFFORD_R: self.__generate_r_cliffords_strategy,
-            StrategyTypes.GENERALIZED_CLIFFORD: self.__generate_generalized_cliffords_strategy
+            StrategyTypes.GENERALIZED_CLIFFORD: self.__generate_generalized_cliffords_strategy,
```

Comment:
I've learned recently that I could use https://docs.python.org/3/library/functools.html#functools.singledispatchmethod for things like this. If your version of Python would allow it, it could be worth checking out.

Reply:
Thanks! I'll give it a read.

Reply:
That's a cool thing, but it won't work here as

> the dispatch happens on the type of the first non-self or non-cls argument

and the factory basically accepts a `experiment_configuration` every time, I believe that current approach is better, but I'll keep this thingy in mind. :)